mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -1818,6 +1817,78 @@ class LowerEqualRewrite : public OpRewritePattern<graphblas::EqualOp> {
   };
 };
 
+class LowerVectorArgMinMaxOpRewrite : public OpRewritePattern<graphblas::VectorArgMinMaxOp> {
+public:
+  using OpRewritePattern<graphblas::VectorArgMinMaxOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(graphblas::VectorArgMinMaxOp op, PatternRewriter &rewriter) const override {
+    // TODO we get seg faults if given a size 0 vector or a sparse vector with no non-zero values.
+    Location loc = op->getLoc();
+
+    Value input = op.vec();
+    RankedTensorType inputType = input.getType().dyn_cast<RankedTensorType>();
+    ArrayRef<int64_t> inputShape = inputType.getShape();
+    
+    Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
+    Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
+    Type indexType = rewriter.getIndexType();
+    Type int64Type = rewriter.getIntegerType(64);
+    Type memref1DI64Type = MemRefType::get({-1}, int64Type);
+    int64_t inputLength = inputShape[0];
+    Value endPosition;
+    if (inputLength == -1) {
+      Type indexType = rewriter.getIndexType();
+      Value pointers = rewriter.create<sparse_tensor::ToPointersOp>(loc, memref1DI64Type, input, c0);
+      Value endPosition64 = rewriter.create<memref::LoadOp>(loc, pointers, c1);
+      endPosition = rewriter.create<IndexCastOp>(loc, endPosition64, indexType);
+    } else {
+      endPosition = rewriter.create<ConstantIndexOp>(loc, inputLength);
+    }
+
+    Type inputElementType = inputType.getElementType();
+    Type memref1DValueType = MemRefType::get({-1}, inputElementType);
+    Value values = rewriter.create<sparse_tensor::ToValuesOp>(loc, memref1DValueType, input);
+    
+    Value initialMax = rewriter.create<memref::LoadOp>(loc, values, c0);
```

Comment:
I should rename the variables using "max" to use "extremum" instead since sometimes it's actually the minimum.

Reply:
https://github.com/metagraph-dev/mlir-graphblas/pull/111/commits/2a90f1977b878dfa50d390c629de204ba929268f addresses this issue.